src/libYARP_dev/src/RGBDSensorParamParser.cpp

```diff
@@ -91,6 +91,7 @@ static bool parseIntrinsic(const Searchable& config, const string& groupName, ya
     }
     Bottle& intrinsic = config.findGroup(groupName);
 
+    realparam.first = "physFocalLength";    realparam.second = &params.physFocalLength; realParams.push_back(realparam);
```

Comment:
This change forces to update the configuration file for running the `depthcamera` driver, since now the group of intrinsic parameters (both RGB and depth) will be marked as malformed since missing this new parameter.
The consequence is that the configuration phase and the open of the device will fail, using old "ini" files.

This problem does not affect the `realsense2` device driver since the intrinsic parameter are not retrieved from configuration file but they are asked to the hardware.

Reply:
Unfortunately, there is not a painless solution for this, the parser has been designed that groups in the configuration files can be optional, but once you add it has to be well-formed, in the sense that if you insert the group of intrinsic parameters you can't forget one parameter.

We have just to decide(cc @barbalberto) if we want to break the user conf files.

Reply:
Ok, I made the change a bit to fast to see it was a mandatory parameter, mainly because it is optional in https://github.com/robotology/yarp/blob/60acf9858220a1475d270dd3ab9f558f35f7d60d/src/libYARP_sig/src/IntrinsicParams.cpp#L44

I thought the change was harmless, but if it is impossible to add a parameter without breaking all the config files, the best would be not to add it (so in the end it is just a change in the documentation from mm to pixel).
I think the physical focal length is not useful.
And even if we add it, I fear that in practice it will end up with the value given in the datasheet of the sensor, which is not the same as a real calibrated value that is not so easy to obtain.

At least that would keep the compatibility with the old config files and people using the focal length in mm in homemade config files could still do so if they really want.

@barbalberto do you really think the physical focal length should be added?

Reply:
I like the PR, so far.
My point is that if we simply change the documentation, we are actually changing the meaning of the parameter. The legacy code wrote with previous meaning will now silently behave badly, when used with device that use the new meaning without warning the user about it.

I believe that having it mandatory is a bit painful because it force the user to update the config file otherwise the program doesn't start, but at the same time it guarantees that, when it does start, it is for sure correct and consistent, which is valuable.

@Nicogene I know that R1 robot files shall be updated, do you know there are other config files around?